### DISCUSSION TITLE
Can you please integrate Eltako FTFSB (A5-04-02) [DONE]

### DISCUSSION THREAD

**Discussion by Ekgev** [mak-gitdev/HA_enoceanmqtt#12] [Request] [answered]
Can you please integrate also Eltako FTFSB (A5-04-02)? Thank you very much!!!!

**Comment by mak-gitdev** [comment, marked as answer]
Ok.

I have an initial support available for your A5-04-02.

As I don't have access to all EnOcean devices, I organized this script so that the end-user can do field tests when requesting a new device support.
For that, it requires that you modify your mapping.yaml file.

Here are the steps to follow:
- First please stop the addon.
- Then copy the [mapping.yaml file](https://github.com/mak-gitdev/HA_enoceanmqtt/blob/support-new-eeps/enoceanmqtt/overlays/homeassistant/mapping.yaml) to your home assistant `/config/mapping.yaml`. Note that `/config/` is the directory where you have your Home Assistant `configuration.yaml` file.
- Then, as you are using the addon, you have a configuration entry named `mapping_file`. Set this entry to `/config/mapping.yaml` and save your configuration.
- Finally, add your A5-04-02 device in your device file then restart the addon.

After that you should see entities corresponding to your device in HA.
Please then report whether the created entities are OK or require modifications, with a picture if possible.


**Note**: There will be 6 entities:
- 1 Humidity and 1 temperature in celcius with full precision
- 1 Humidity and 1 temperature in celcius with 1 decimal
- 1 temperature in fahrenheit
- 1 status indicating if the device is available or not

**Reply by Ekgev** [reply]
Many thanks! I just made a quick test, unfortunately there appear no values.
Instead the log Shows
2022-11-23 18:18:48,688 INFO: Device enoceanmqtt/Temperature_sensors/Vorgarten (@: 05914D26 / EEP: A5-04-02) updated on device database
2022-11-23 18:19:34,112 WARNING: Cannot find type in EEP!
2022-11-23 18:19:34,113 WARNING: message not interpretable: enoceanmqtt/Temperature_sensors/Vorgarten
Is this helpful for you to find the issue or do you need additional Information?
Thanks

**Reply by mak-gitdev** [reply]
Oops, you're right !

I have to modify some code that is not accessible outside the addon like mapping.yaml.
So for the moment, you just have entities with no values.

I have to push a new version of the addon.
Tonight or tomorrow...

**Reply by mak-gitdev** [reply]
Hi,

A new version of the addon is available.
See #17 for more details.

**Reply by Ekgev** [reply]
Hi, thank you very much. It works with FTFSB!!!!

**Reply by mak-gitdev** [reply]
Great !
May I see a screenshot of the device page in HA ?
Thanks

**Reply by Ekgev** [reply]
Now I will try to integrate the other enocean devices from the "old" smarthome system.

**Comment by mak-gitdev** [comment]
Hi,

Sorry I wasn't clear enough.
I just wanted to see the device page of one of your A5-04-02 to see how it is represented. To see if the created entities are as I imagined.

**Reply by Ekgev** [reply]
Do you mean this:
![image0](https://user-images.githubusercontent.com/118467264/204581375-b0b7d91e-d322-467e-9917-dd3f80cc8adc.png)

**Reply by mak-gitdev** [reply]
Yes exactly.
It seems that some entities are not OK.

e2m_xxx_status is undefined and there should be one entity returning the temperature in fahrenheit.

Is the `mapping_file` entry in the configuration tab empty ? If not, please leave it empty and restart the addon. Do not forget to refresh the device page to update the entities.

Thanks !

**Reply by Ekgev** [reply]
I checked this again. The mapping file entry is empty.

**Reply by Ekgev** [reply]
Can I help with further information of tests?

**Reply by mak-gitdev** [reply]
Yes please.

Could you repeat the steps previously described at https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/12#discussioncomment-4216479 and just replace the line 505 of mapping.yaml:
```yaml
                   {% if value_json.TSN == 0 %}off{% else %}on{% endif %}
```
by:
```yaml
                   {% if value_json.TSN == 0 %}OFF{% else %}ON{% endif %}
```

Thanks !

**Reply by Ekgev** [reply]
The result is
![grafik](https://user-images.githubusercontent.com/118467264/204887922-6a1d83e1-9168-45d9-a983-89d754ffef1e.png)

**Reply by Ekgev** [reply]
After additional reboot
![grafik](https://user-images.githubusercontent.com/118467264/204888566-27e0ea95-7392-4af4-bc0b-9b5b548f4bcc.png)

**Reply by mak-gitdev** [reply]
Ok now status is OK.
Thanks ;)
